File: python/leetcode/majority-element-ii.py
class Solution:
    def majorityElement(self, nums):
        seen = {}
        if len(nums) == 1:
            return nums
        majority_count = len(nums) // 3
        for n in nums:
            if n in seen:
                seen[n] += 1
            else:
                seen[n] = 1
        return [k for k, v in seen.items() if v > majority_count]

File: java/leetcode/majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
      HashMap<Integer, Integer> seen = new HashMap<Integer, Integer>();
      List<Integer> list = new ArrayList<Integer>();
      if (nums.length == 1) {
          list.add(nums[0]);
          return list;
      }
      int majorityCount = nums.length / 3;
      for (int i = 0; i < nums.length; i++) {
          seen.put(nums[i], seen.getOrDefault(nums[i], 0) + 1);
      }

      for (int i : seen.keySet()) {
        if (seen.get(i) > majorityCount) {
          list.add(i);
        }
      }
      return list;
    }
}
